src/main/java/com/scalar/cassy/transferer/FileSystemFileUploader.java

```diff
@@ -0,0 +1,83 @@
+package com.scalar.cassy.transferer;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.inject.Inject;
+import com.palantir.giraffe.host.HostControlSystem;
+import com.scalar.cassy.config.BackupConfig;
+import com.scalar.cassy.exception.FileTransferException;
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Future;
+import org.apache.commons.lang3.time.DurationFormatUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class FileSystemFileUploader extends FileSystemTransfererBase implements FileUploader {
+  private static final Logger logger = LoggerFactory.getLogger(FileSystemFileUploader.class);
+  private final HostControlSystem hostConnection;
+
+  @Inject
+  public FileSystemFileUploader(HostControlSystem hostConnection) {
+    this.hostConnection = hostConnection;
+  }
+
+  @Override
+  public Future<Void> upload(Path file, String key, String storageBaseUri) {
```

Comment:
I don't think we should change the existing interface. I think it is better to inject location object in the constructor instead.

Maybe a container that has `HostControlSystem` and `storageBaseUri` ?
(GiraffeUtil.openSshConnection() has already accepts the URI of `storageBaseUri` so it seems it can be simplified easily.)

Reply:
Ok, as you suggest I plan to modify the GiraffeUtil like this.
```
public class RemoteStorageSshConnectionManager {
  private final HostControlSystem hostControlSystem;
  private final String storagePath;

  public RemoteStorageSshConnectioManager(URI storageBaseURI) throws IOException {
    Preconditions.checkArgument(storageBaseURI.getScheme().equals("ssh"));
    String userName =
        storageBaseURI.getUserInfo() != null
            ? storageBaseURI.getUserInfo()
            : System.getProperty("user.name");
    SshCredential credential =
        PublicKeySshCredential.fromFile(
            userName, Paths.get(System.getProperty("user.home"), ".ssh/id_rsa"));
    SshHostAccessor sshHostAccessor =
        SshHostAccessor.forCredential(Host.fromHostname(storageBaseURI.getHost()), 22, credential);
    this.hostControlSystem = sshHostAccessor.open();
    this.storagePath = storageBaseURI.getPath();
  }

  public static HostControlSystem openSshConnection(URI sshURI) throws IOException {}

  public HostControlSystem getHostControlSystem() {
    return hostControlSystem;
  }

  public String getStoragePath() {
    return storagePath;
  }
}
```
And instead of injecting the `HostControlSystem` inside the `FileSystemFileDownloader` and `FileSystemFileUploader`, I will inject an instance of `RemoteStorageSshConnectioManager`